collie/model/hybrid_matrix_factorization.py

```diff
@@ -493,31 +515,41 @@ def forward(self, users: torch.tensor, items: torch.tensor) -> torch.tensor:
                 + self.item_biases(items)
             )
         else:
+            self._calculate_metadata_output(
+                metadata_type='user',
+                users=users,
+                items=items
+            )
+            self._calculate_metadata_output(
+                metadata_type='item',
+                users=users,
+                items=items
+            )
             # TODO: remove self.device and let lightning do it
-            item_metadata_output = self.item_metadata[items, :].to(self.device)
-            if self.item_metadata_layers is not None:
-                for metadata_nn_layer in self.item_metadata_layers:
-                    item_metadata_output = self.dropout(
-                        F.leaky_relu(
-                            metadata_nn_layer(item_metadata_output)
-                        )
-                    )
+            #item_metadata_output = self.item_metadata[items, :].to(self.device)
+            #if self.item_metadata_layers is not None:
+            #    for metadata_nn_layer in self.item_metadata_layers:
+            #        item_metadata_output = self.dropout(
+            #            F.leaky_relu(
+            #                metadata_nn_layer(item_metadata_output)
+            #            )
+            #        )
 
             # TODO: remove self.device and let lightning do it
-            user_metadata_output = self.user_metadata[users, :].to(self.device)
-            if self.user_metadata_layers is not None:
-                for metadata_nn_layer in self.user_metadata_layers:
-                    user_metadata_output = self.dropout(
-                        F.leaky_relu(
-                            metadata_nn_layer(user_metadata_output)
-                        )
-                    )
+            #user_metadata_output = self.user_metadata[users, :].to(self.device)
+            #if self.user_metadata_layers is not None:
+            #    for metadata_nn_layer in self.user_metadata_layers:
+            #        user_metadata_output = self.dropout(
+            #            F.leaky_relu(
+            #                metadata_nn_layer(user_metadata_output)
+            #            )
+            #        )
 
             # TODO: make this matrix factorization instead of only a MLP
             combined_output = torch.cat((self.user_embeddings(users),
                                          self.item_embeddings(items),
-                                         item_metadata_output,
-                                         user_metadata_output), 1)
+                                         self.item_metadata_output,
+                                         self.user_metadata_output), 1)
```

Comment:
Can we have this output be with the order `user_metadata, user_embedding, item_embedding, item_metadata`?

Reply:
👍